### PR TITLE
Cleanup: cleaning up .first/.second [source/common/config]

### PR DESCRIPTION
**Pull request by stedsome**
This is the second PR to this https://github.com/envoyproxy/envoy/issues/12354. This time is in the common/config submodule. 

Commit Message: 
Additional Description:
Risk Level: Low, solely cosmetic changes. 
Testing: All existing tests passed. 
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by stedsome** [comment]
@junr03 @asraa this has been out here for a while now. Do you mind taking a look?

**Comment by junr03** [comment]
@asraa carries context from #12354 so I will let her take this.

File: source/common/config/new_grpc_mux_impl.cc [hunk 1]
```diff
@@ -73,8 +73,8 @@ void NewGrpcMuxImpl::onDiscoveryResponse(
 }
 
 void NewGrpcMuxImpl::onStreamEstablished() {
-  for (auto& sub : subscriptions_) {
-    sub.second->sub_state_.markStreamFresh();
+  for (auto& [type_url, sub] : subscriptions_) {
```
Comment (asraa) [inline source/common/config/new_grpc_mux_impl.cc:76]: nit: suggest full name `subscription`?

Comment (stedsome) [reply source/common/config/new_grpc_mux_impl.cc]: thanks, just changed it.

**Pull request merged**